jmbitcoin/jmbitcoin/secp256k1_transaction.py

```diff
@@ -9,825 +9,181 @@
-    if not isinstance(script, bytes):
-        script = binascii.unhexlify(script)
-    if scriptCode is not None and not isinstance(scriptCode, bytes):
-        scriptCode = binascii.unhexlify(scriptCode)
-    if isinstance(sig, bytes):
-        sig = binascii.hexlify(sig).decode('ascii')
-    if isinstance(pub, bytes):
-        pub = binascii.hexlify(pub).decode('ascii')
+    # script verification flags
+    flags = set()
 
-    hashcode = binascii.unhexlify(sig[-2:])
+    def return_err(e):
+        return None, "Error in signing: " + repr(e)
 
-    if amount:
-        modtx = segwit_signature_form(deserialize(tx), i,
-                    scriptCode, amount, hashcode, decoder_func=lambda x: x)
-    else:
-        modtx = signature_form(tx, i, script, hashcode)
-    return ecdsa_tx_verify(modtx, sig, pub, hashcode)
-
-
-def sign(tx, i, priv, hashcode=SIGHASH_ALL, usenonce=None, amount=None,
-         native=False):
-    """
-    Given a serialized transaction tx, an input index i, and a privkey
-    in bytes or hex, returns a serialized transaction, in hex always,
-    into which the signature and/or witness has been inserted. The field
-    `amount` flags whether segwit signing is to be done, and the field
-    `native` flags that native segwit p2wpkh signing is to be done. Note
-    that signing multisig is to be done with the alternative functions
-    multisign or p2wsh_multisign (and non N of N multisig scripthash
-    signing is not currently supported).
-    """
-    if isinstance(tx, basestring) and not isinstance(tx, bytes):
-        tx = binascii.unhexlify(tx)
-    if len(priv) <= 33:
-        priv = safe_hexlify(priv)
-    if amount:
-        serobj = p2wpkh_sign(tx, i, priv, amount, hashcode=hashcode,
-                               usenonce=usenonce, native=native)
-    else:
-        pub = privkey_to_pubkey(priv, True)
-        address = pubkey_to_address(pub)
-        signing_tx = signature_form(tx, i, mk_pubkey_script(address), hashcode)
-        sig = ecdsa_tx_sign(signing_tx, priv, hashcode, usenonce=usenonce)
-        txobj = deserialize(tx)
-        txobj["ins"][i]["script"] = serialize_script([sig, pub])
-        serobj = serialize(txobj)
-    if isinstance(serobj, basestring) and isinstance(serobj, bytes):
-        return binascii.hexlify(serobj).decode('ascii')
-    else:
-        return serobj
-
-def p2wpkh_sign(tx, i, priv, amount, hashcode=SIGHASH_ALL, native=False,
-                usenonce=None):
-    """Given a serialized transaction, index, private key in hex,
-    amount in satoshis and optionally hashcode, return the serialized
-    transaction containing a signature and witness for this input; it's
-    assumed that the input is of type pay-to-witness-pubkey-hash.
-    If native is False, it's treated as p2sh nested.
-    """
+    assert isinstance(tx, CMutableTransaction)
+    # using direct local access to libsecp256k1 binding, because
+    # python-bitcoinlib uses OpenSSL key management:
```

Comment:
Since this PR is about bitcointx, this comment seems to be obsolete, OpenSSL in bitcointx is option and used only for `CPubKey.verify_nonstrict()`

Reply:
Agreed, addressed in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/536/commits/1e2fc3848295a05f4ee685fd435f277c378acadc